src/components/VisualizeEditorWindow/worker.js

```diff
@@ -0,0 +1,19 @@
+
+
+import * as druid from "@saehrimnir/druidjs";
+
+self.onmessage = function (e) {
+    let data1 = JSON.parse(e.data);
+    let data = [];
+    data1?.result?.points?.map((point) => {
+        data.push(point.vector);
+    });
+    if (data.length) {
+        const D = new druid["TSNE"](data, {}); // ex  params = { perplexity : 50,epsilon :5}
+        let next = D.generator();//default = 500 iterations 
+        for (const i of next) {
+            console.log(typeof i);
+            self.postMessage(i);
```

Comment:
Do you think we could limit `self.postMessage(i);` based on the time between operations? Otherwise it might be re-rendering too fast. Let's say only do `postMessage` each 100ms, but still keep the generator running